src/pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -191,16 +190,27 @@ def setup(self, trainer: "pl.Trainer") -> None:
 
         self.setup_precision_plugin()
 
+    def setup_optimizers(self, trainer: "pl.Trainer") -> None:
+        invalid_params_error = False
+        try:
+            super().setup_optimizers(trainer)
+        except ValueError as e:
+            if "optimizer got an empty parameter list" not in str(e):
+                raise
+            invalid_params_error = True
```

Comment:
You don't need this flag, the block below can be indented into this except block

And then use `raise ValueError(...) from e`

Reply:
actually, we do.. even if the try block is successful, there's another condition where this error will be raised.

Reply:
Sorry, thought that had an `and`, not an `or`